File: client/app/types/product.ts
import { StaticImageData } from "next/image";

export interface Good {
  id: string | number;
  image: string | StaticImageData;
  title: string;
  rating: number;
  price: number;
  currency: Currency;
  discount?: number;
  category?: string;
}

export type Currency = "USD" | "RUB" | "EUR";

export interface Product {
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt: string | null;
  name: string;
  description: string;
  price: number;
  currency: Currency;
  quantity: number;
}

File: server/models/product.go
package models

import (
	"gorm.io/gorm"
)

type Product struct {
	gorm.Model
	Name        string  `json:"name" binding:"required"`
	Description string  `json:"description" binding:"required"`
	Price       float64 `json:"price" binding:"required"`
	Currency    string  `json:"currency"`
	Quantity    int     `json:"quantity" binding:"required"`
}
